src/Misc.ino

```diff
@@ -1601,6 +1592,21 @@ void addToLog(byte logLevel, const char *line)
 }
 
 
+/********************************************************************************************\
+  Clean up all before going to sleep or reboot.
+  \*********************************************************************************************/
+void prepareShutdown()
+{
+  runPeriodicalMQTT();  // Flush outstanding MQTT messages
+  process_serialWriteBuffer();
+  flushAndDisconnectAllClients();
+  saveUserVarToRTC();
+  delay(100); // give the node time to flush all before reboot or sleep
+  saveToRTC();
+  SPIFFS.end();
+}
+
+
```

Comment:
what about delay(100) after Spiffs.end() ?

Reply:
Is it needed?
If so (or the slightest doubt) I will add it ofcourse :)

Reply:
just move it from line 1604 after SPIFFS.end() :)